cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -777,3 +777,24 @@ func TestImagePullCheck(t *testing.T) {
 		t.Fatalf("expected 2 errors but got %d: %q", len(errors), errors)
 	}
 }
+
+func TestNumCPUCheck(t *testing.T) {
+	var tests = []struct {
+		numCPU      int
+		numErrors   int
+		numWarnings int
+	}{
+		{0, 0, 0},
+		{int(^uint(0) >> 1), 1, 0},
+	}
+
+	for _, rt := range tests {
```

Comment:
FYI we gravitate towards using subtests with `t.Run` for new code, but not critical here.